enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,216 @@
+3. **SNO (Single Node OpenShift)**: The same nodes host both the ingress and API components. Both FQDNs (`api` and wildcard) resolve to the same IP, making the challenge feasible.
+
+To address this gap, a small proxy was developed. This proxy runs on the cluster as a DaemonSet (control plane nodes) and then adds iptables rules to the nodes and ensures that connections reaching the API on port 80 are redirected to the OpenShift Ingress Routers. The proxy implementation creates a reverse proxy to the apps VIP and uses `nftables` to redirect traffic from `API:80` to `PROXY:8888`.
+
+- **Proxy Code**: [GitHub Repository](https://github.com/mvazquezc/cert-mgr-http01-proxy/tree/main)
+- **Deployment Manifest**: [Manifest Link](https://github.com/mvazquezc/cert-mgr-http01-proxy/blob/main/manifests/deploy-in-ocp.yaml)
+
+This enhancement aims to provide a robust solution for managing certificates for the API endpoint in baremetal environments.
+
+### User Stories
+
+1. **As a cluster administrator**, I want to manage custom certificates for the API endpoint (`api.cluster.example.com`) using an external ACME CA, so that I can ensure secure communication for my cluster's API.
+2. **As a cluster administrator on a baremetal platform**, I want a reliable solution to handle HTTP01 challenges for the API endpoint, even when the endpoint is not managed by OpenShift Ingress, so that I can avoid manual workarounds.
+3. **As a developer**, I want a simple deployment mechanism for the HTTP01 challenge proxy, so that I can easily integrate it into my existing cluster setup.
+
+### Goals
+
+- Provide a reliable mechanism for Cert Manager to complete HTTP01 challenges for the API endpoint (`api.cluster.example.com`) in baremetal environments.
+- Ensure compatibility with various OpenShift topologies, including Standard Clusters, Compact Clusters, and SNO.
+- Minimize operational complexity by using a DaemonSet-based deployment and `nftables` for traffic redirection.
+
+## Proposal
+
+The HTTP01 Challenge Proxy will be implemented via DaemonSet running on the cluster. It will:
+
+- Redirect HTTP traffic from the API endpoint (`api.cluster.example.com`) on port 80 to the OpenShift Ingress Routers.
+- Use `nftables` for traffic redirection from `API:80` to `PROXY:8888`.
+- Be deployed using a manifest that includes all necessary configurations.
+
+The proxy will ensure compatibility with various OCP topologies, including SNO, MNO, and Compact clusters, addressing the challenges of HTTP01 validation for the API endpoint.
+
+### API Extensions
+
+A new CR type may be created and can be applied to clusters.  This new typed will be stored in the [openshift/api](https://github.com/openshift/api) repo.
+
+Potential Example of a CR:
+
+```
+apiVersion: network.openshift.io/v1alpha1
+kind: HTTP01ChallengeProxy
+metadata:
+  name: example-http01challengeproxy
+  namespace: default
+spec:
+  # Add fields here to specify the desired state of the HTTP01ChallengeProxy
```

Comment:
The motivation and non-goals states only one endpoint, `api.cluster.example.com`, is scoped for this work. Forward-thinking, is there any other endpoints that users may want to be able to proxy that we would want to support?

What other things might a user want to configure here, either now or long-term?

Reply:
That's the only user-exposed endpoint exposed by the platform that isn't routed through the ingress controller, I don't think we need to support other endpoints.

Reply:
One thing I noticed is that there is specification around the proxy being hardcoded to port `8888` - why this port specifically? How should users handle a scenario where something else that is critical to their system needs to run on port `8888`? Would it be worth making the port a configurable value?